### ISSUE TITLE
Dsolve shouldn't return series solutions

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#15916]
For most ODE inputs `dsolve` returns an exact solution or raises NotImpementedError. Occasionally though it returns a series solution e.g.:
```julia
In [32]: dsolve(f(x).diff(x) + sin(x*f(x)**2))                                                                                                 
Out[32]: 
              2  2     3  4 ⎛  3    ⎞        
            C₁ ⋅x    C₁ ⋅x ⋅⎝C₁  + 6⎠    ⎛ 6⎞
f(x) = C₁ - ────── + ──────────────── + O⎝x ⎠
              2             24               
```
Speaking just for myself I never want this solution when calling `dsolve`. It takes a long time to compute these solutions and there isn't any way of knowing in advance that this will happen.

I think it would be better if obtaining power series solutions was handled by a separate API.

**Comment by asmeurer** [comment]
Agree, it shouldn't do it unless you use `series=True`, or even a separate function `dsolve_series`.

**Comment by asmeurer** [comment]
> there isn't any way of knowing in advance that this will happen.

Technically you can check `classify_ode` first. But regardless I still think dsolve should only return exact solutions by default.

**Comment by divyanshu132** [comment]
I have checked `classify_ode` and  `_desolve`. I found that whenever `classify_ode` predicts `hint = "1st_power_series"` than we will have the series solution.

Let's take a simple example:
`dsolve(Eq(y.diff(x), (x+1)/(2-y)))`

![sympy2](https://user-images.githubusercontent.com/36084189/53862085-22414980-400c-11e9-98d9-eb7b7120a39a.png)


@asmeurer @oscarbenjamin should we modify the `hint`??

**Comment by oscarbenjamin** [comment]
I think that ideally there would be a separate function for getting series solutions but it's not easy to split it out of dsolve right now.

I guess some people are using this so it needs to go through deprecation.

I suggest to add a new keyword argument `series_solutions=` to dsolve. This argument should default to None in which case it will do the current behaviour but it should warn if that results in series solutions that obtaining series solutions in that way is deprecated. There should also be a warning if the user provides arguments x0 or n since those should be removed ultimately.

If the user passes False then it should not give series solutions (this should become the default after deprecation).

Otherwise the series_solutions argument can be a dict of values relevant to series solutions (n, x0, etc).
